src/features/filter.ts

```diff
@@ -5,8 +5,20 @@
   status: 'all',
 };
 
-export const filterSlice = createSlice({
+const filterSlice = createSlice({
   name: 'filter',
   initialState,
-  reducers: {},
+  reducers: {
+    setFilter: (value, action) => {
+      return { status: action.payload, query: value.query };
+    },
+
+    setQuery: (value, action) => {
+      return { status: value.query, query: action.payload };
```

Comment:
In the `setQuery` reducer, the `query` is correctly set from `action.payload`, but the `status` should remain unchanged. The current implementation incorrectly assigns `value.query` to `status`, which is incorrect. Ensure that `status` is not altered in this reducer.